### PR TITLE
JAMES-2855 Upgrade reactor, feign

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[9e02152695cdb8746339b861d307d6227bcd7014] 10:54:57.675 [ERROR] r.c.s.Schedulers - Scheduler worker in group main failed with an uncaught exception
[9e02152695cdb8746339b861d307d6227bcd7014] java.lang.NoSuchMethodError: com.google.common.util.concurrent.Futures.addCallback(Lcom/google/common/util/concurrent/ListenableFuture;Lcom/google/common/util/concurrent/FutureCallback;)V
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.blobstore.strategy.internal.DeleteAllKeysInList.deleteBlobsAndEmptyDirs(DeleteAllKeysInList.java:274)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.blobstore.strategy.internal.DeleteAllKeysInList.executeOneIteration(DeleteAllKeysInList.java:342)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.blobstore.strategy.internal.DeleteAllKeysInList.execute(DeleteAllKeysInList.java:413)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.blobstore.util.internal.BlobUtilsImpl.clearContainer(BlobUtilsImpl.java:77)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.blobstore.internal.BaseBlobStore.clearContainer(BaseBlobStore.java:192)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.blobstore.internal.BaseBlobStore.clearContainer(BaseBlobStore.java:181)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.s3.blobstore.S3BlobStore$1.apply(S3BlobStore.java:189)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.s3.blobstore.S3BlobStore$1.apply(S3BlobStore.java:186)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.util.Predicates2$RetryablePredicate.apply(Predicates2.java:117)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.s3.blobstore.S3BlobStore.deletePathAndEnsureGone(S3BlobStore.java:195)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at org.jclouds.blobstore.internal.BaseBlobStore.deleteContainer(BaseBlobStore.java:228)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:190)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:121)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at reactor.core.publisher.FluxFilterFuseable$FilterFuseableSubscriber.tryOnNext(FluxFilterFuseable.java:143)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at reactor.core.publisher.FluxPublishOn$PublishOnConditionalSubscriber.runSync(FluxPublishOn.java:800)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at reactor.core.publisher.FluxPublishOn$PublishOnConditionalSubscriber.run(FluxPublishOn.java:936)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[9e02152695cdb8746339b861d307d6227bcd7014] 	at java.lang.Thread.run(Thread.java:748)
```

**Comment by Arsnael** [comment]
It seems you have a new version for feign that came out this month, 10.3.0 -> https://mvnrepository.com/artifact/io.github.openfeign/feign-core

Please upgrade :)

File: server/blob/blob-objectstorage/pom.xml [hunk 1]
```diff
@@ -85,6 +85,7 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
+            <version>25.1-jre</version>
```
Comment (chibenwa) [inline server/blob/blob-objectstorage/pom.xml]: That's cheating! 

We need a bit of analysis here: **which dependency is pulling and relying on guava 25 dependency?** -> jcloud-core does...

Comment (trantienduchn) [reply server/blob/blob-objectstorage/pom.xml]: that was the jclouds dependency, we encountered this problem before and I saw that was what we did: overriding guava version. (mpt/impl/imap-mailbox/cyrus/pom.xml and mailbox/zoo-seq-provider/pom.xml).

Comment (chibenwa) [reply server/blob/blob-objectstorage/pom.xml]: These are not used in any "final" guice product - hence this solution is not applicable

Comment (trantienduchn) [reply server/blob/blob-objectstorage/pom.xml]: okay, I put the guava upgrade into an separated PR https://github.com/linagora/james-project/pull/2580. And try to find a way

**Comment by chibenwa** [comment]
:green_apple: before CI reboot

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Pull request closed without merging**